src/test/java/org/openrewrite/staticanalysis/SimplifyConstantTernaryExecutionTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+@SuppressWarnings("ConstantConditionalExpression")
+public class SimplifyConstantTernaryExecutionTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new SimplifyConstantTernaryExecution());
+    }
+
+    @Test
+    @DocumentExample
+    void simplifyConstantTernaryExecution() {
+        // Given
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  void test() {
+                      System.out.println(true ? "foo" : "bar");
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void test() {
+                      System.out.println("foo");
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void simplifyUnnecessaryParenthesesTernary() {
+        // Given
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  void test() {
+                      System.out.println((true) ? "foo" : "bar");
+                  }
+              }
+              """,
+            """
+              class Test {
+                  void test() {
+                      System.out.println("foo");
```

Comment:
@knutwannheden this test is failing and is currently generating a change that looks like this:

```java
System.out.println( "foo");
```
instead of the expected:
```java
System.out.println("foo");
```

Same issue with the test above. Not sure what's going on here. Is this something broken with the `AutoFormat` visitor?

Reply:
Could be. I can investigate.

Reply:
The "problem" here is that what gets auto-formatted is the true or false branch of the ternary only. In an of itself it isn't incorrectly formatted and removing the whitespace prefix would require checking what it is contained in.

While it might be possible to change how auto-formatting works (although that currently seems quite difficult, since our cursor doesn't know which "feature" of the containing element is currently being visited), I think we should here instead just format the containing element. I've updated the PR accordingly.